### PR TITLE
fix(statue): fix statue not update

### PR DESCRIPTION
**Pull request by wangcanfengxs**
Signed-off-by: wangcanfeng <wangcanfeng@corp.netease.com>

### CODE REVIEW

File: controllers/goharbor/harborcluster/ctrl_setup.go [hunk 1]
```diff
@@ -123,3 +144,12 @@ var harborClusterPredicateFuncs = predicate.Funcs{
 		return true
 	},
 }
+
+func (r *Reconciler) CRDInstalled(ctx context.Context, client dynamic.Interface, crdName string) bool {
+	_, err := client.Resource(apiextensions.Resource("customresourcedefinitions").WithVersion("v1")).Get(ctx, "tenants.minio.min.io", metav1.GetOptions{})
```
Comment (heww) [inline controllers/goharbor/harborcluster/ctrl_setup.go:149]: `"tenants.minio.min.io"` => crdName

File: manifests/cluster/deployment.yaml [hunk 2]
```diff
@@ -13429,6 +13429,14 @@ rules:
...
   - update
   - watch
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - get
+  - list
+
```
Comment (heww) [inline manifests/cluster/deployment.yaml:13439]: These permissions must be defined by the kubebuilder and generated by the kustomize.

File: controllers/goharbor/harborcluster/ctrl_setup.go [hunk 3]
```diff
@@ -86,10 +92,25 @@ func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager) err
...
 
-	return ctrl.NewControllerManagedBy(mgr).
+	builder := ctrl.NewControllerManagedBy(mgr).
 		For(&goharborv1alpha2.HarborCluster{}).
-		WithEventFilter(harborClusterPredicateFuncs).
-		Complete(r)
+		Owns(&goharborv1alpha2.Harbor{}).
+		WithEventFilter(harborClusterPredicateFuncs)
+
+	if r.CRDInstalled(ctx, dClient, "tenants.minio.min.io") {
```
Comment (steven-zou) [inline controllers/goharbor/harborcluster/ctrl_setup.go:100]: Suggest defining const for each CRD full name

File: manifests/cluster/deployment.yaml (hunk 2 above)
Comment (steven-zou) [reply manifests/cluster/deployment.yaml:13439]: I think canfeng has already added permission kubebuilder annotations and this is the regenerating result.

Comment (steven-zou) [reply manifests/cluster/deployment.yaml:13439]: the generated files are ignored by git

cc @heww 

@wangcanfengxs  please confirm it.

**Review by steven-zou — COMMENTED** [review]
comments left

File: manifests/cluster/deployment.yaml (hunk 2 above)
Comment (wangcanfengxs) [reply manifests/cluster/deployment.yaml:13439]: `manifests/cluster/deployment.yaml` should update.
`config/rbac/role.yaml` will regenerate, but ignored by git

**Review by steven-zou — APPROVED** [review]
LGTM

**Review by heww — APPROVED** [review]

**Pull request merged**